fiftyone/core/view.py

```diff
@@ -546,13 +559,13 @@ def _iter_samples(self):
                 yield sample
 
     def _make_sample(self, d):
-        if getattr(self, "_make_sample_fcn", None) is None:
+        if self._make_sample_fcn is None:
```

Comment:
IDK why I used `getattr` and `delattr` for these variables originally, but now that they're declared in `__init__`, I believe this change was necessary: https://github.com/voxel51/fiftyone/pull/4885/commits/4dc4525104c64188d029b7ae0c01deae69d594a0

Reply:
nvm, tests failed with https://github.com/voxel51/fiftyone/pull/4885/commits/4dc4525104c64188d029b7ae0c01deae69d594a0 and now I remember why `getattr` and `delattr` are used: because subclasses also need to use these variables but can't explicitly call `super().__init__()` due to usage of dunder variables like `__name`.

https://github.com/voxel51/fiftyone/pull/4885/commits/247328374307a25fca2a6e4b42dafd47beb178f0 reverts my change from https://github.com/voxel51/fiftyone/pull/4885/commits/4dc4525104c64188d029b7ae0c01deae69d594a0 and also removes the explicit declaration of `_make_sample_fcn` and `_make_frame_fcn` so that `DatasetView` and its subclasses are consistent in *not* explicitly declaring them.

Reply:
I see, though I don't fully understand. It is a little "smelly" that the variables are declared in slots but not initialized. But that is the way it was before.
I'll go with it.